connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/RestServerTest.java

```diff
@@ -174,6 +174,14 @@ public void testAdvertisedUri() {
         config = new DistributedConfig(configMap);
         server = new RestServer(config);
         Assert.assertEquals("http://my-hostname:8080/", server.advertisedUrl().toString());
+
+        // correct listener is chosen when http and https are configured and advertised listener is http
```

Comment:
Nit: perhaps mention that the order of the listeners config is important:
```suggestion
        // correct listener is chosen when https listener is configured before http listener and advertised listener is http
```

Reply:
Ack, added.